indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -0,0 +1,593 @@
+        required: false
+    }, {
+        type: 'input',
+        name: 'monthReminder',
+        label: Translate.string('Send Booking reminders X days before (monthly)'),
+        required: false
+    }], [{
+        type: 'header',
+        label: 'Custom Attributes'
+    }, {
+        type: 'attributes',
+        placeholder: Translate.string('Add new attributes'),
+    }, {
+        type: 'header',
+        label: Translate.string('Equipment Available')
+    }, {
+        type: 'equipment',
+        placeholder: Translate.string('Add new equipment')
+    }, {
+        type: 'header',
+        label: Translate.string('Nonbookable Periods'),
+    }, {
+        type: 'nonBookablePeriods'
+    }, {
+        type: 'message',
+        content: Translate.string('Room has been successfully edited.')
+    }]
+];
+
+
+class RoomEditModal extends React.Component {
+    static propTypes = {
+        equipmentTypes: PropTypes.arrayOf(PropTypes.object).isRequired,
+        favoriteUsers: PropTypes.array.isRequired,
+        onClose: PropTypes.func.isRequired,
+        roomId: PropTypes.number.isRequired,
+    };
+
+    state = {
+        attributes: null,
+        room: null,
+        roomAttributes: null,
+        roomAvailability: null,
+        roomEquipment: null,
+        submitSucceeded: false,
+    };
+
+    componentDidMount() {
+        this.fetchDetailedRoom();
+        this.fetchRoomAttributes();
+        this.fetchAttributes();
+        this.fetchRoomAvailability();
+        this.fetchRoomEquipment();
+    }
+
+    async fetchDetailedRoom() {
+        const {roomId} = this.props;
+        let response;
+        try {
+            response = await indicoAxios.get(fetchRoomURL({room_id: roomId}));
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({room: camelizeKeys(response.data)});
+    }
+
+    async fetchRoomAttributes() {
+        const {roomId} = this.props;
+        let response;
+        try {
+            response = await indicoAxios.get(fetchRoomAttributesURL({room_id: roomId}));
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({roomAttributes: response.data});
+    }
+
+    async fetchAttributes() {
+        let response;
+        try {
+            response = await indicoAxios.get(fetchAttributesURL());
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        this.setState({attributes: response.data});
+    }
+
+
+    async fetchRoomAvailability() {
+        const {roomId} = this.props;
+        let response;
+        try {
+            response = await indicoAxios.get(fetchRoomAvailabilityURL({room_id: roomId}));
+        } catch (error) {
+            handleAxiosError(error);
+            return;
+        }
+        const roomAvailability = camelizeKeys(response.data);
+        roomAvailability.nonbookablePeriods.map(o => {
```

Comment:
Out of curiosity, why is it called `o`? Couldn't be called `period`?

Reply:
or at least the usual `x` when mapping stuff and the name doesn't matter.

btw, you call `map()` without using the return value ==> use `forEach()` instead